### PR TITLE
add time out assert data collection and presentation

### PR DESCRIPTION
**Pull request by altendky**
### Purpose:

Adds time out assertion data collection and presentation similar to what was added for benchmarks.

https://github.com/Chia-Network/chia-blockchain/actions/runs/6884258354?pr=16831

I am personally comfortable leaving the remaining `TODO:`s in place for later improvements.

### Testing Notes:

Top 50 Time Out Asserts: https://github.com/Chia-Network/chia-blockchain/actions/runs/6898367177?pr=16831#summary-18769328079
Benchmarks: https://github.com/Chia-Network/chia-blockchain/actions/runs/6898367139?pr=16831#summary-18768254817

### Draft For:
- [x] https://github.com/Chia-Network/chia-blockchain/pull/16846

### CODE REVIEW

**Review by aqk — DISMISSED** [review]

**Review by emlowe — DISMISSED** [review]

**Comment by emlowe** [comment]
Coverage-diff exception.

aside, maybe `tests/process_junit.py` should be coverage exempted.

File: tests/process_junit.py [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+                "max": max(self.durations),
+                "mean": mean(self.durations),
+            },
+        }
+
+    def link(self, prefix: str, line_separator: str) -> str:
+        return f"{prefix}{self.file_path.as_posix()}{line_separator}{self.line}"
+
+
+def sub(matchobj: re.Match[str]) -> str:
```
Comment (arvidn) [inline tests/process_junit.py:51]: I don't see this function used anywhere

**Review by arvidn — COMMENTED** [review]
seems reasonable

File: tests/process_junit.py (hunk 1 above)
Comment (altendky) [reply tests/process_junit.py:51]: Yep, indeed.  Seems I made this monstrosity pointless.  Thanks.

Comment (altendky) [reply tests/process_junit.py]: Removed in https://github.com/Chia-Network/chia-blockchain/pull/16831/commits/de39fc06bfd154308455936d22c4c5a7e5ce4d19.

**Review by emlowe — DISMISSED** [review]

**Comment by emlowe** [comment]
coverage diff exemption approved

**Review by emlowe — DISMISSED** [review]

**Review by emlowe — DISMISSED** [review]

**Review by emlowe — APPROVED** [review]

**Review by cmmarslender — APPROVED** [review]

**Pull request merged**